### PR TITLE
Fix issues with CONTRIBUTING.md

### PR DESCRIPTION
**Pull request by faheel**
With these changes, I have:
-  [x] Improved formatting.
-  [x] Added links in contents to all headings.
-  [x] Fixed some links, and added a few.
-  [x] Fixed some minor grammatical and syntax errors.

This closes issue #86.

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -1,106 +1,77 @@
...
-### Table of contents
+## Contents
 
-What should I know before I get started?
+### Before you get started
 
-- code of conduct
-- decisions around how we built mathsteps
-- mathJS expression trees
+- [Code of Conduct](#code-of-conduct)
```
Comment (evykassirer) [inline CONTRIBUTING.md]: these links are great, thanks

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -1,106 +1,77 @@
...
-  find it in 'flattenOperands.js'
+### How (and why) we built mathsteps
+
+Read about how and why we built mathsteps on our [blog](https://blog.socratic.org/stepping-into-math-open-sourcing-our-step-by-step-solver-9b5da066ae36).
+
+### Expression trees in mathJS
+
+Most of this code iterates over expression trees to make step by step simplifications. We use [mathJS expresison trees](http://mathjs.org/docs/expressions/expression_trees.html#expression-trees), which we recommend you learn a bit about.
+
+There are a few different types of nodes that show up in the tree. This stepper uses OperationNode, ParenthesisNode, ConstantNode, SymbolNode and FunctionNode. You can also read about them on the [mathJS expressions documentation](http://mathjs.org/docs/expressions/expression_trees.html#nodes).
```
Comment (evykassirer) [inline CONTRIBUTING.md]: please add the oxford comma after `SymbolNode` back :p

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -1,106 +1,77 @@
...
-  So at most points in the codebase, there should be no operators with sign `-`
-  If you're curious what the code that modifies subtraction looks like, you can
-  find it in 'flattenOperands.js'
+### How (and why) we built mathsteps
+
+Read about how and why we built mathsteps on our [blog](https://blog.socratic.org/stepping-into-math-open-sourcing-our-step-by-step-solver-9b5da066ae36).
+
+### Expression trees in mathJS
+
+Most of this code iterates over expression trees to make step by step simplifications. We use [mathJS expresison trees](http://mathjs.org/docs/expressions/expression_trees.html#expression-trees), which we recommend you learn a bit about.
```
Comment (evykassirer) [inline CONTRIBUTING.md]: haha I forgot to add the link! thanks for fixing

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -1,106 +1,77 @@
...
-  someone else can reproduce it.
-- **Contribute code!** We'd love to have more contributors working on this. Check
-  out the section below with more information on how to contribute, and feel
-  free to email mathsteps@socratic.org with any questions!
+- **Spread the word!** If you think mathsteps is cool, tell your friends! Let them know they can use this and that they can contribute.
+- **Suggest features!** Have an idea for something mathsteps should solve or a way for it to teach math better? If your idea is not an [existing issue](https://github.com/socraticorg/mathsteps/issues?q=is%3Aopen+is%3Aissue+label%3Aenhancement), create a new issue with the label "enhancement".
+- **Report bugs!** If the bug is not an [existing issue](https://github.com/socraticorg/mathsteps/issues?q=is%3Aopen+is%3Aissue+label%3Abug), create a new issue with the label "bug" and provide as many details as you can, so that someone else can reproduce it.
+- **Contribute code!** We'd love to have more contributors working on this. Check out the section below with more information on how to contribute, and feel free to mail us at mathsteps@socratic.org with any questions!
```
Comment (evykassirer) [inline CONTRIBUTING.md]: mail -> email

File: CONTRIBUTING.md [hunk 5]
```diff
@@ -1,106 +1,77 @@
...
-  a [new one](https://github.com/socraticorg/mathsteps/issues/new)
-- make sure all the unit tests pass (with `npm test`) before creating the
-  pull request, and please add your own tests for what you changed as well.
-  If you're not sure how to add tests or are confused about why tests are
-  failing, it's also fine to create the pull request first and we'll help
-  you get things working.
+We're excited to see your [pull request](https://help.github.com/articles/about-pull-requests/)!
 
-### Testing
+- If you want to work on something, please comment on the [related issue](https://github.com/socraticorg/mathsteps/issues) on GitHub before you get started, so others are aware that you're working on it. In case there's is no existing issue for the change you'd like to make, you can [create a new issue](https://github.com/socraticorg/mathsteps/issues/new).
```
Comment (evykassirer) [inline CONTRIBUTING.md]: "In case there's" -> "In the case that there's"

(or "If there's no existing...")

Comment (faheel) [inline CONTRIBUTING.md]: :heavy_check_mark:

File: CONTRIBUTING.md (hunk 4 above)
Comment (faheel) [inline CONTRIBUTING.md]: :heavy_check_mark:

File: CONTRIBUTING.md (hunk 2 above)
Comment (faheel) [inline CONTRIBUTING.md]: Learned something new! :smile:

**Comment by faheel** [comment]
Just to make things clear, in the above commits I've only [modified CONTRIBUTING.md](https://github.com/socraticorg/mathsteps/pull/85/files#diff-6a3371457528722a734f3c51d9238c13), and merged the changes that were pushed to `master` since my first commit.

**Comment by evykassirer** [comment]
Can you get the rest of the changes out of the diff by merging in master? The same issue happened in #68 and merging master seemed to fix it.

File: CONTRIBUTING.md (hunk 2 above)
Comment (evykassirer) [inline CONTRIBUTING.md]: yay oxford commas ^_^

**Comment by evykassirer** [comment]
I made a gist for what I think the file should look like with the 80 char max

https://gist.github.com/evykassirer/a3c10666b9c7e7232abe478204521422

Mainly, I think it's better to not have line breaks between sentences and stuff. This is based off of what I've seen for markdown files, but I'm definitely not an expert on this and if you think what you have is nicer I'm open to conversation.

Thanks for dealing with my pickiness ^__^ I'm very grateful for this changes and excited to have a nicer thing for new contributors to read :D

**Comment by faheel** [comment]
Done! :relieved:

**Comment by evykassirer** [comment]
perfect, the merge commit worked :)

thanks for your patience and for this change! wooo 🎉

**Pull request merged**